File: STRMaterial.m
classdef STRMaterial
    properties
        Id;
        Name;
        E; %Young's Modulus
        G;
        mu; % possion Ration
    end

    methods
        function obj = STRMaterial(id, name, e, p)
            obj.Id = id;
            obj.Name = name;
            obj.E = e;
            obj.mu = p;
        end
        
        function ToString(obj)
            fprintf('Material (%s) #%i\n', obj.Name, obj.Id);
        end
    end
end

File: src/STRMaterial.cpp
#include "STRMaterial.hpp"
#include <iostream>

STRMaterial::STRMaterial(int id, const std::string& name, double e, double g)
    : Id(id), Name(name), E(e), G(g) {
}

void STRMaterial::ToString() const {
    std::cout << "Material (" << Name << ") #" << Id << "\n";
}
